tests/cypress/support/commands.ts

```diff
@@ -709,3 +715,108 @@ Cypress.Commands.add('compareClusterResourceCount', (clusterName) => {
+  cy.contains('.title', 'Feature Flags').should('be.visible');
+
+  cy.filterInSearchBox(flagName);
+  cy.verifyTableRow(0, 'Disabled', flagName);
+  cy.open3dotsMenu(flagName, 'Activate');
+
+  // Activate the Feature Flag
+  cy.get('div.card-title')
+  .should('be.visible')
+  .then(($el) => {
+    if ($el.text().includes('Are you sure?')) {
+      // Check that 'Are you sure?' is visible before activating.
+      cy.checkModalCardTitle('Are you sure?', true, false);
+      cy.clickButton("Activate");
+    }
+  });
+
+  // Check that 'Are you sure?' is not visible anymore.
+  cy.checkModalCardTitle('Are you sure?', false, false);
+
+  // After activation, Rancher pod will restart.
+  // Wait for 'Waiting for Restart' to disappear.
+  cy.checkModalCardTitle('Waiting for Restart', true)
+})
+
+Cypress.Commands.add('checkModalCardTitle', (expectedText, shouldHaveText, waitForRestart=true) => {
+  // Pop-up menu comes under <div modal>, when interacting with different
+  // dialog boxes. Check the text present/absense on each Modal page Title.
+  cy.get('div.card-title')
+    .should('be.visible')
+
+  cy.get('div.card-title')
+  .invoke('text') // Get the raw text
+  .then((text) => {
+    // Normalize the text by replacing non-breaking spaces with regular spaces and trimming
+    const normalizedText = text.replace(/\u00A0/g, ' ').trim();
+
+    if (shouldHaveText) {
+      expect(normalizedText).to.equal(expectedText); 
+    } else {
+      expect(normalizedText).to.not.equal(expectedText); 
+    }
+
+    // Now check for the "Waiting for Restart" condition if waitForRestart is true
+    if (waitForRestart) {
+      cy.contains(expectedText, { timeout: 180000 }).should('not.exist');
+    }
+  });
+})
+
+Cypress.Commands.add('moveClusterToWorkspace', (clusterName, workspaceName) => {
+  // Navigate to Clusters page when other navigation is present.
+  cy.get('body').then((body) => {
+    if (body.find('.title').text().includes('Clusters')) {
+      return true
+    }
+    else {
+      cy.accesMenuSelection('Continuous Delivery', 'Clusters');
+    }
+  })
+
+  cy.filterInSearchBox(clusterName);
+  cy.verifyTableRow(0, 'Active', clusterName);
+  cy.open3dotsMenu(clusterName, 'Change workspace')
+
+  // Check pop-up menu has title and no need to wait for restart Rancher(false).
+  cy.checkModalCardTitle('Assign Cluster To…', true, false);
+
+  cy.get(
+    `[data-testid="workspace_options"] .vs__dropdown-toggle, 
+    .labeled-select.edit.hoverable [aria-label="Search for option"]`
+  ).click();
+  cy.contains(workspaceName, { matchCase: false })
+    .should('be.visible')
+    .click();
+  cy.clickButton('Apply');
+
+  // It automatically switches to Newly created workspace.
+  // We will change the Fleet workspace explicitly to newly created workspace
+  cy.fleetNamespaceToggle(workspaceName);
+  cy.clickNavMenu(['Clusters']);
+  cy.filterInSearchBox(clusterName);
+  cy.verifyTableRow(0, 'Wait Check-In', clusterName);
+  cy.wait(30000);
```

Comment:
Is this needed?
`verifyTableRow` contains an implicit wait of 60 seconds.
If for some reason it is not working, this could be replaced by another implicit wait on the active status.

Reply:
Yes, we have to see the Cluster status i.e. "Wait Check-In", just to ensure that cluster is moving to another workspace.

Reply:
Then, "if for some reason it is not working, this could be replaced by another implicit wait on the active status."
Do you think this can be tried?

Reply:
I will give a try to it.

Reply:
I will remove this `Wait Check-In` I can see It will wait for Active state.